Unity-Technologies/com.unity.multiplayer.samples.coop
# feat: replace guid with auth playerid for session management

### Description (*)

This PR replaces our custom GUID it with the PlayerId obtained from UGS's AuthenticationService when signed in. When not signed in, if using direct IP connection, it will use our custom GUID. It also removes the workaround we had to allow clients with the same GUID to connect on debug builds, since we are using profiles for that. Now the arguments used to set profiles are also used to update our custom GUID when connecting to a host to differentiate each client if using multiple builds or parrelsync clones.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2640](https://jira.unity3d.com/browse/MTT-2640)
### Manual testing scenarios

1. Start a game with two or more clients
2. Disconnect each client
3. Reconnect to the game with each client
4. See that each client is reconnected with the right character and the right data
5. Do it again through direct IP while signed in and not signed in.
### Questions or comments

### Contribution checklist
